js/ooyala_ssai.js

```diff
@@ -366,11 +368,21 @@ OO.Ads.manager(function()
     this.resumeAd = (ad) => {
       if (adMode) {
         _handleTrackingUrls(this.currentAd, ["resume"]);
-        if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && isFinite(ad.duration)) {
+        if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && _.isFinite(ad.duration)) {
+          var duration = ad.duration;
+          //Deducting the already played duration of ad  from the actual ad duration for making timer accurate
+          if(_.isFinite(this.adIdDictionary[ad.ad.data.id].startTime) && _.isFinite(this.adIdDictionary[ad.ad.data.id].pauseTime) )
```

Comment:
it's better to use destructuring to increase code readability
`const { startTime, pauseTime } = this.adIdDictionary[ad.ad.data.id];`

Reply:
Incorporated the changes.